### ISSUE TITLE
support for http(s) base apps

### ISSUE THREAD

**Issue by vencax** [MutonUfoAI/pgina#17]
Hi, I am absolute newbee to pgina core code but I would greatly appreciate that instead of configuring of multiple plugins I can point pgina to certain http(s) endpoint that would provide all necessary info about given credential. I.E. all necessary logic is offloaded into 3rd party app (could be written in any language in any way - freedom). [The idea is quite old](https://github.com/vencax/windows-rest-auth).
Would it be easy or problematic? Appreciate any feedback. THanks :)

**Comment by MutonUfoAI** [comment]
I don't know what you want!
Are you talking about configuration or authentication ....

**Comment by vencax** [comment]
Hi, sorry, maybe I have expressed myself wrong. My idea is that pgina core grab user credentials and send http(s) request. Response (HR) to such request shall contain if the given credentials exists, is allowed to login, its group permissions, etc. The content of HR must be well defined to allow pgina to react on in accordingly. 
In other words instead of configuration of mysql, postgresql, whatever sql, ldap, whatever plugin, you simply configure http(s) endpoint. Leaving all sql,ldap, whatever logic handling on http app on that endpoint.
Is it more clear now?

**Comment by MutonUfoAI** [comment]
Sure that can be done
I have absolutely no need for it but feel free to add it.....

**Comment by MutonUfoAI** [comment]
I see you've started your project :)
If you have questions feel free to ask ....

**Comment by vencax** [comment]
Yeah, I have just finished what I want from the plugin. Compiled dll and expecting 2be happy :)

**Comment by MutonUfoAI** [comment]
glad to see that you continue on your own plugin
to be honest i had no time to look at it
A documentation would be fine too :)

**Comment by vencax** [comment]
Hi, documentation is in README.md within plugin folder. Is there another place for that?

**Comment by MutonUfoAI** [comment]
explain it to someone who doesnt know what this plugin is good for
.)what is it good for
.)how does it work internally
.)whats the server background

**Comment by MutonUfoAI** [comment]
I've made some changes to your plugin
please verify.

Some questions:
whats the purpose of _getTxtRecords
why are you sending a plain password

why not using JSON Web Token?
https://github.com/jwt-dotnet/jwt

**Comment by vencax** [comment]
- _getTxtRecords shall resolve DNS txt record that shall contain URL of the http endpoint (SE) where to send requests
- sending plain info because I expected the SE secured via HTTPS
- using JWT is possible, just like any other hashing facility to prevent plain passwords, but now it is as simple as possible

**Comment by vencax** [comment]
UPDATE:
- change password implementation
- error handling done (hopefully) right (there was issues with login to existing local profiles when no netwrk connection was awailable)
- configuration priority order: environment variable -> DNS record -> ginaSetting -> default

Now it is complete (at least for my requirements). Feel free to close this issue, or comment.
Cheers!
